### PR TITLE
feat(cash): /references finds payment uuid by ref

### PR DESCRIPTION
**Pull request by jniles**
This commit adds an API route to find a payment UUID by the human
readable reference.  Use it like this:
1. `GET /cash/references/:reference` - returns the uuid of a cash
   payment matching the reference if it exists.  Otherwise, returns a 404
   NOT FOUND error.

There are accompanying integration tests to demonstrate functionality.

### CODE REVIEW

File: server/controllers/finance/cash.js [hunk 1]
```diff
@@ -249,15 +267,41 @@ exports.update = function update(req, res, next) {
...
   'use strict';
   // TODO
   next();
-};
+}
+
+
+/**
+ * retrieves cash payment uuids from a reference string (e.g. HBB123)
+ * GET /cash/references/:reference
```
Comment (DedrickEnc) [inline server/controllers/finance/cash.js:286]: Can you use `jsdoc` for comment.

Comment (jniles) [inline server/controllers/finance/cash.js:286]: This is `jsdoc` syntax.  The `@description` is optional if the first line is the description ([src](http://usejsdoc.org/tags-description.html)).  Can you be more specific about what you'd like to see?

**Comment by DedrickEnc** [comment]
@jniles 
Using the jsdoc is not a big deal, I can merge it when ever. it looks good for me

**Comment by jniles** [comment]
Okay.  I'm happy to add whatever comments you think are necessary.  Otherwise, you can merge it.

File: server/controllers/finance/cash.js (hunk 1 above)
Comment (DedrickEnc) [inline server/controllers/finance/cash.js:286]: `@method` or `@function` , `@name`,  `@param` , `@description`, `@example` are missed.

**Pull request merged**